dotnet/src/Connectors/Connectors.Google/Extensions/VertexAIKernelBuilderExtensions.cs

```diff
@@ -37,7 +37,7 @@ public static class VertexAIKernelBuilderExtensions
     public static IKernelBuilder AddVertexAIGeminiChatCompletion(
         this IKernelBuilder builder,
         string modelId,
-        Func<Task<string>> bearerTokenProvider,
+        Func<ValueTask<string>> bearerTokenProvider,
```

Comment:
Is this not a breaking change, considering that this method and class is public?

Reply:
I'm curious why this is even needed. The key can't just be provided to this as it is for other providers?

As for being a breaking change, it is, but I believe the whole library is still marked experimental.

Reply:
THis is needed because the VertexAI API's are not ApiKey based they use a Authentication Token with expiration date, that token normally have a lifetime of 30 minutes or less, and a provider was the ideal option to delegate this implementation.

Google AI API's on the other hand support ApiKey approach and we don't have that constructor option.

Reply:
@westey-m Answering your question, it is but as @stephentoub also pointed this is an experimental project and we have some liberty to make those.